### PR TITLE
Companion PR for #7039: grandpa-rpc dont share subscription manager, only executor

### PR DESCRIPTION
**Pull request by octol**
Companion PR for #7039.
Don't share subscription manager with grandpa-rpc, it is enough to only share the executor.

### CODE REVIEW

File: rpc/src/lib.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ use sc_client_api::light::{Fetcher, RemoteBlockchain};
 use sc_consensus_babe::Epoch;
 use sp_block_builder::BlockBuilder;
 pub use sc_rpc::DenyUnsafe;
-pub use jsonrpc_pubsub::manager::SubscriptionManager;
+pub use sc_rpc::SubscriptionTaskExecutor;
```
Comment (tomusdrw) [inline rpc/src/lib.rs:33]: These two could be merged.

**Review by tomusdrw — APPROVED** [review]
lgtm!

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -158,7 +158,10 @@ fn new_partial<RuntimeApi, Executor>(config: &mut Configuration) -> Result<
 		consensus_common::DefaultImportQueue<Block, FullClient<RuntimeApi, Executor>>,
 		sc_transaction_pool::FullPool<Block, FullClient<RuntimeApi, Executor>>,
 		(
-			impl Fn(polkadot_rpc::DenyUnsafe, polkadot_rpc::SubscriptionManager) -> polkadot_rpc::RpcExtension,
+			impl Fn(
+				polkadot_rpc::DenyUnsafe,
+				polkadot_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline node/service/src/lib.rs:163]:
```suggestion
				polkadot_rpc::SubscriptionTaskExecutor,
```

File: service/src/lib.rs [hunk 3]
```diff
@@ -121,7 +121,10 @@ pub fn new_partial<RuntimeApi, Executor>(config: &mut Configuration, test: bool)
 		consensus_common::DefaultImportQueue<Block, FullClient<RuntimeApi, Executor>>,
 		sc_transaction_pool::FullPool<Block, FullClient<RuntimeApi, Executor>>,
 		(
-			impl Fn(polkadot_rpc::DenyUnsafe, polkadot_rpc::SubscriptionManager) -> polkadot_rpc::RpcExtension,
+			impl Fn(
+				polkadot_rpc::DenyUnsafe,
+				polkadot_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline service/src/lib.rs:126]:
```suggestion
				polkadot_rpc::SubscriptionTaskExecutor,
```

**Review by andresilva — APPROVED** [review]
lgtm

**Comment by andresilva** [comment]
bot merge

**Pull request merged**